src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -857,6 +858,11 @@ internal unsafe int GetNonRandomizedHashCodeOrdinalIgnoreCase()
 
                 while (length > 2)
                 {
+                    if (!Utf16Utility.AllCharsInUInt32AreAscii(ptr[0]) || !Utf16Utility.AllCharsInUInt32AreAscii(ptr[1]))
```

Comment:
It might be faster to:

```cs
uint a = ptr[0];
uint b = ptr[1];
if (!Utf16Utility.AllCharsInUInt32AreAscii(a | b)) { goto NotAscii; }
// use already-cached locals 'a' and 'b' instead of derefing ptr again
```